examples/TeensyRDMController/TeensyRDMController.ino

```diff
@@ -0,0 +1,144 @@
+#include <TeensyDmx.h>
+#include <rdm.h>
+
+#define DMX_REDE 2
+
+// It was an easy job to register a manufacturer ID to myself as explained
+// on http://tsp.esta.org/tsp/working_groups/CP/mfctrIDs.php.
+// The first two bytes are the manufacturer ID, the next four must be unique
+// for every device produced by that manufacturer.
+// An easy way to do this is to use TeensyMAC to fetch the Teensy's unique
+// MAC address, then map each OUI to a value in the third byte and use the
+// three unique bytes of the MAC for bytes 4-6.
+// The ID below is designated as a prototyping ID.
+byte myUid[] = {0x7f, 0xf0, 0x20, 0x12, 0x00, 0x00};
+
+// Broadcast to all devices, broadcasts don't generate a response
+byte theirUid[] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
+
+enum Status { UN_MUTE, UN_MUTE_IDLE, DISCOVERY, DISCOVERY_IDLE, IDENTIFY_ON, IDENTIFY_ON_IDLE, IDENTIFY_OFF, IDENTIFY_OFF_IDLE };
+Status currentState = UN_MUTE;
+unsigned long lastAction = 0;
+
+void discoveryComplete(CallbackStatus callbackStatus, byte *uids, uint32_t uidCount) {
+  if (callbackStatus == CallbackStatus::CB_SUCCESS) {
+    Serial.println("Printing UIDs:");
+
+    Serial.print("Length: ");
+    Serial.println(uidCount);
+
+    for(uint32_t i = 0; i < uidCount; i++)
+    {
+      for(int j = 0; j < RDM_UID_LENGTH; j++)
+      {
+        Serial.print(uids[((i * RDM_UID_LENGTH) + j)], HEX);
+        Serial.print(":");
+      }
+      Serial.println("");
+    }
+    Serial.println("");
+    memcpy(theirUid, uids, RDM_UID_LENGTH);
+    currentState = IDENTIFY_ON;
+  } else {
+    Serial.println("Discovery CB failed");
+  }
+}
+
+void printRdm(CallbackStatus callbackStatus, RdmData *data) {
+  if (callbackStatus == CallbackStatus::CB_SUCCESS) {
+    Serial.println("Decoding RDM:");
+    // TODO(Peter): Handle NAcks
+
+    Serial.print("Length: ");
+    Serial.println(data->length);
+    Serial.print("Data Length: ");
+    Serial.println(data->dataLength);
+    
+    Serial.print("Data: ");
+    for(int i = 0; i < data->dataLength; i++)
+    {
+       Serial.print(char(data->data[i]));
+    }
+    Serial.println("");
+  } else {
+    Serial.println("RDM CB failed");
+  }
+}
+
+struct RdmInit rdmData {
+  myUid,
+  0x00000100,
+  "TeensyDMX v0.1",
+  "Teensyduino",
+  1,  // Device ID
+  "DMX Node",
+  E120_PRODUCT_CATEGORY_DIMMER_CS_LED,
+  1,  // The DMX footprint
+  1,  // The DMX startAddress - only used for RDM
+  0,  // Additional commands length for RDM
+  0,   // Definition of additional commands
+  &discoveryComplete,
+  &printRdm
+};
+
+TeensyDmx Dmx(Serial1, &rdmData, DMX_REDE);
+
+byte DMXVal[] = {50};
+
+byte lowerUid[] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
```

Comment:
These seem like useful constants to have as static members of the class?  I'd be reluctant to bloat the class for those not using RDM, but I believe as static they will get compiled out if not used.

Reply:
Something still on the TODO list is the discovery algorithm should be handled in the library, so there's one version with all the minor corner cases covered. As soon as that's done, this UID becomes irrelevant.

Reply:
Ok